app/models/post.rb

```diff
@@ -22,4 +24,29 @@ def tweet_content
     max_title_length = 140 - url.length
     title[0...max_title_length] + url
   end
+
+  def tags_string
+    tags.map(&:title).join(", ")
+  end
+
+  def tags_string=(value)
+    @tags_list = []
+    value.strip.downcase.split(/, *| +/).each do |tag|
+      @tags_list.push(tag.strip)
+    end
+    @tags_list = @tags_list.uniq
+  end
+
+  def create_tags_from_tag_string
+    tags.clear
+    @tags_list.each do |tag_title|
+      existing_tag = Tag.find_by(title: tag_title)
```

Comment:
You can use `find_or_create_by` to write less code. http://apidock.com/rails/v4.2.1/ActiveRecord/Relation/find_or_create_by